[WIP] Feature - Welcome Screen

arthurdenner opened the pull request:
Hello, everybody. 

This PR is not finished, as the title says, but I wanted to share what I've built so far to get feedback and to ask a question regarding #22.

So, I started working on this feature with the components that already existed, but after a while, I started to not like the result, so I decided to start _almost_ from the scratch.

So far, I've added a simple page with some information about the project (pure copy & paste from GitHub and the website) and the forms for sign in and sign up. The following things are missing:

- [x] language options: I still need to see how to get the languages available. It shouldn't be hard, I just haven't done it yet;
- [x] passwordConfirm validation: There is a problem in Yup to make this validation, but [they're working on it](https://github.com/jaredpalmer/formik/issues/90#issuecomment-357071930);
- [ ] integration with the API: I'm waiting on the API to be live so I can finish this.

My question is:

- Are we going to support translation for the welcome screen? I feel like we should, but since we don't have all the symbols proofread, maybe we shouldn't rely in automatic translation for this, I don't know.

That's it for now. Please feel free to comment and leave your opinion about the UI or anything else.

arthurdenner committed:
refactor(welcome-screen): Adding information and access buttons

arthurdenner committed:
refactor(welcome-screen): Adding information and access buttons

arthurdenner committed:
feat(welcome-screen): Adding SignUp and Login pages

arthurdenner committed:
feat(welcome-screen): Adding SignUp and Login pages

arthurdenner committed:
refactor(welcome-screen): Wrapping Login and Signup in Dialogs

arthurdenner committed:
refactor(welcome-screen): Wrapping Login and Signup in Dialogs

arthurdenner committed:
refactor(welcome-screen): Adding missing fields and tweaking styles again :laughing:

arthurdenner committed:
refactor(welcome-screen): Adding missing fields and tweaking styles again :laughing:

arthurdenner committed:
feat(welcome-screen): Adding reducer to handle firstVisit and finishing "Go to app" button

arthurdenner committed:
feat(welcome-screen): Adding reducer to handle firstVisit and finishing "Go to app" button

arthurdenner committed:
chore(welcome-screen): Removing old Intro folder

arthurdenner committed:
chore(welcome-screen): Removing old Intro folder

shayc reviewed (changes requested):
Good PR! regarding translation, if you want I can do it.. let me know 😃

arthurdenner committed:
refactor(welcome-screen): Fixing action type naming

arthurdenner committed:
refactor(welcome-screen): Fixing action type naming

arthurdenner committed:
style(welcome-screen): Cleaning CSS and adding more semantic naming

arthurdenner committed:
style(welcome-screen): Cleaning CSS and adding more semantic naming

arthurdenner committed:
chore(welcome-screen): Adding language options to the form

arthurdenner committed:
chore(welcome-screen): Adding language options to the form

arthurdenner committed:
feat(welcome-screen): Adding PropTypes to new components

arthurdenner committed:
feat(welcome-screen): Adding PropTypes to new components

arthurdenner committed:
Merge branch 'master' into feature/welcome-screen-v2

shayc committed:
Merge branch 'feature/welcome-screen-v2' of https://github.com/arthurdenner/cboard into arthurdenner-feature/welcome-screen-v2

shayc committed:
Merge branch 'arthurdenner-feature/welcome-screen-v2' of https://github.com/shayc/cboard into arthurdenner-feature/welcome-screen-v2

shayc committed:
fix: assorted visual/text changes

shayc committed:
wip: inital translation support

shayc committed:
add: translations

arthurdenner committed:
Merge pull request #1 from shayc/arthurdenner-feature/welcome-screen-v2

Adding translations and tweaking styles

arthurdenner committed:
feat(welcome-screen): Starting to integrate with the API
A constants file was created to determine the API URL in development and production
axios was installed to handle the API calls nicely
It was created a LoadingIcon component based on the MUI CircularProgress
Login and Signup were moved to .component files
Some fields in the Signup form were removed because they're not in the user model at the moment
Were created containers and actions for the Login and Signup
The App reducer was modified to contain loading status for Login and Signup
